lib/salesforce/database.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require 'salesforce/donation_creator'
+require 'salesforce/supporter_creator'
+require 'salesforce/supporter_finder'
+
+module Salesforce
+  class Database
+    def self.add_donation(data)
+      new(data).add_donation
+    end
+
+    def initialize(data)
+      @data = data
+    end
+
+    def add_donation
+      return [:missing_email] unless email_present?
+      return supporter_result.errors unless supporter_result.okay?
+      donation_creation_errors
+    end
+
+    private
+
+    attr_reader :data
+
+    def email_present?
+      data.respond_to?(:email) && data.email
```

Comment:
These validators are only about validating presence, so that `data.foo` doesn't blow up because `data` is `nil`, or SF blows up because a required field is `nil`. Malformed field validation is done in the form before submission.